### ISSUE TITLE
False positive for java.lang.correctness.eqeq.eqeq

### ISSUE THREAD

**Issue by Sjord** [returntocorp/sgrep-rules#4019]
```
BooleanExpression expr = null;
BooleanExpression other = null;
expr = (expr != null ? expr.and(other) : other);
```

```
$ semgrep -c 'r/java.lang.correctness.eqeq' test.java
┌────────────────┐
│ 1 Code Finding │
└────────────────┘
                            
    test.java
   ❯❯❱ java.lang.correctness.eqeq.eqeq
          ❰❰ Blocking ❱❱
          `expr == expr` or `expr != expr` is always true. (Unless the value compared is a float or double).
          To test if `expr` is not-a-number, use `Double.isNaN(expr)`.                                      
          Details: https://sg.run/qx1x                                                                      
                                                                                                            
            3┆ expr = (expr != null ? expr.and(other) : other);

                
                
┌──────────────┐
│ Scan Summary │
└──────────────┘
✅ Scan completed successfully.
 • Findings: 1 (1 blocking)
 • Rules run: 1
 • Targets scanned: 1
 • Parsed lines: ~100.0%
 • No ignore information available
Ran 1 rule on 1 file: 1 finding.
```
